### ISSUE TITLE
Unused variable reported inside for() loop with only two expressions.

### ISSUE THREAD

**Issue by nicoSWD** [scrutinizer-ci/php-analyzer#690]
The analyser reports that the variable `$operator` is not used, and that I can remove the assignment. What it apparently doesn't see is, that it's being used inside a loop, and that it's indeed necessary.

Here's the code in question:
```php
private function evalGroup(array $group)
{
    $flag = \null;
    $operator = \null;

    for ($offset = 0; isset($group[1][$offset]);) {
        $value = $group[1][$offset++];

        if ($value === '&' || $value === '|') {
            $operator = $value;
        } elseif ($value === '1' || $value === '0') {
            if (!isset($flag)) {
                $flag = (int) $value;
            } elseif ($operator === '&') {
                $flag &= $value;
            } else {
                $flag |= $value;
            }
        } else {
            throw new Exceptions\EvaluatorException(sprintf(
                'Unexpected "%s"',
                $value
            ));
        }
    }

    return $flag;
}
```

I'm pretty sure it has something to do with `for`'s third (optional) expression. When I changed the code to this:

```php
// ...

for ($offset = 0; isset($group[1][$offset]); $offset++) {
    $value = $group[1][$offset];

    // ...
}
```
... the issue was reported to be fixed.

![screen shot 2015-07-23 at 19 01 20](https://cloud.githubusercontent.com/assets/205852/8856776/3ff67420-316d-11e5-9849-52aad780bfb3.png)

see https://scrutinizer-ci.com/g/nicoSWD/php-rule-parser/inspections/6505ba6e-0c48-42a8-a918-b92fd36649b6/issues/files/src/nicoSWD/Rules/Evaluator.php?status=new&orderField=path&order=asc&issueId=26c056b2-2374-4088-a661-722f3c9843cc

https://scrutinizer-ci.com/g/nicoSWD/php-rule-parser/inspections/f5fec1f6-ed52-4e8d-8245-786477bb0e58/issues/?status=fixed
